src/App.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from 'react'
+import axios from 'axios'
+import View from './components/View'
+import Cards from './components/Cards'
+import { IData } from './models/dataModel.js'
+import './App.css'
+
+function App() {
+    const [newCustomerView, setnewCustomerView] = useState<boolean>(false)
+    const [data, setData] = useState<IData[] | undefined>()
+
+    const changeView = (): void => {
+        setnewCustomerView(!newCustomerView)
+    }
+
+    useEffect((): void => {
```

Comment:
This useEffect have two responsibilities in one:

1. Make fetch request to the server
2. Sort and set data depends on current view

In our realities we have no need to make second fetch call after change of current view and we can split our logic on single responsibilities effects. This allow to us make only one fetch call on App initialisation and process data on demand.

```tsx
const [data, setData] = useState<IData[]>([]);
const [processedData, setProcessedData] = useState<IData[]>([]);
const [isNewCustomerView, setIsNewCustomerView] = useState<boolean>(false);

const fetchData = async (): Promise<IData[]> => {
    const response = await axios.get<IData[]>('https://www.mocky.io/v2/5bc3b9cc30000012007586b7');
    return response.data;
};

const processData = (data: IData[]): IData[] => {
    if (isNewCustomerView) {
        return data
            .filter((promo) => promo.onlyNewCustomers)
            .sort((a, b) => a.sequence - b.sequence);
    } else {
        return [...data].sort((a, b) => a.sequence - b.sequence);
    }
};

useEffect(() => {
    const initializeData = async (): Promise<void> => {
        try {
            const fetchedData = await fetchData();
            setData(fetchedData);
        } catch (error: any) {
            console.error("Error fetching data:", error);
        }
    };

    initializeData();
}, []);  // Empty dependency array to run only on mount

useEffect(() => {
    const updatedData = processData(data);
    setProcessedData(updatedData);
}, [isNewCustomerView, data]);  // Run when isNewCustomerView or data changes

// Note: Use 'processedData' for rendering in the component and handle the 'error' state in the UI.
```

Reply:
This was very helpful!
Although I had to put " useState<IData[] | undefined>(); "  for both _data_ and _processedData_
I hope that is not an issue 😊

Reply:
Also there was still a warning:
**React Hook useEffect has a missing dependency: 'processData'. Either include it or remove the dependency array  react-hooks/exhaustive-deps**
So I added it to useEffect as dependency and made processData a useCallback function with a [isNewCustomerView] dependency.
It works fine and I hope that makes sense. 😊